src/Microsoft.Azure.ServiceBus/Message.cs

```diff
@@ -6,1136 +6,202 @@ namespace Microsoft.Azure.ServiceBus
-                }
-                else
-                {
-                    this.initializedMembers |= MessageMembers.To;
-                }
-            }
-        }
-
-        /// <summary>Specifies whether the message has been consumed.</summary>
-        /// <value>true if the message has been consumed; otherwise, false.</value>
-        public bool IsBodyConsumed
-        {
-            get
-            {
-                // Body is consumed in 2 cases.
-                // - Someone called GetBody() - this happens in both send and receive.
-                // - Send operation "considered" a message been sent.
-                // Note: this is not a thread safe boolean. IsBodyConsumed == false
-                // can still lead to Send/GetBody() throwing exception saying body is consumed
-                // in a multi-threaded scenario.
-                this.ThrowIfDisposed();
-                return this.getBodyCalled == 1 || this.messageConsumed == 1;
-            }
-        }
-
-        /// <summary>Specifies if message is a received message or not.</summary>
-        public bool IsReceived => this.receiverHeaders != null;
-
-        internal Guid LockTokenGuid
-        {
-            get
-            {
-                this.ThrowIfDisposed();
-                this.ThrowIfNotLocked();
-                return this.receiverHeaders.LockToken;
-            }
-
-            set
-            {
-                this.ThrowIfDisposed();
-                this.EnsureReceiverHeaders();
-
-                this.receiverHeaders.LockToken = value;
-                if (value != Guid.Empty)
-                {
-                    this.initializedMembers |= MessageMembers.LockToken;
-                }
-                else
-                {
-                    this.ClearInitializedMember(MessageMembers.LockToken);
-                }
-            }
+        public Message()
+            : this(default(byte[]))
+        {
         }
 
-        internal short PartitionId
+        /// <summary>
+        /// Creates a new Message using a string that is UTF8 encoded.
+        /// </summary>
+        /// <param name="body">The body of the message.</param>
+        public Message(string body)
+            : this(Encoding.UTF8.GetBytes(body))
         {
-            get
-            {
-                this.ThrowIfDisposed();
-                return this.partitionId;
-            }
-
-            set
-            {
-                if (value < 0)
-                {
-                    throw Fx.Exception.AsError(new ArgumentOutOfRangeException(nameof(this.PartitionId)));
-                }
+        }
 
-                this.ThrowIfDisposed();
-                this.initializedMembers |= MessageMembers.PartitionId;
-                this.partitionId = value;
-            }
+        public Message(byte[] array)
+        {
+            this.Body = array;
+            this.SystemProperties = new SystemPropertiesCollection();
```

Comment:
So if we keep it in the same file and rename the class to just SystemProperties there is a name clash between that and the property on Message. I am of the opinion to keep the class in the file, and therefore the name would stay the same. I personally like the clarity of 'this class is only ever used in here'